pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -164,11 +164,10 @@ func (p *DefaultProvider) createAMIOptions(ctx context.Context, nodeClass *v1bet
 		return nil, err
 	}
 	// Get constrained security groups
-	securityGroups, err := p.securityGroupProvider.List(ctx, nodeClass)
-	if err != nil {
-		return nil, err
+	if nodeClass.Generation != nodeClass.Status.SecurityGroupStatus.ObservedGeneration {
+		return nil, fmt.Errorf("waiting until nodeclass is updated")
```

Comment:
The error should describe the error, not the handling of the error
```suggestion
		return nil, fmt.Errorf("detected spec-status generation mismatch")
```